mongofiles/mongofiles.go

```diff
@@ -467,8 +500,21 @@ func (mf *MongoFiles) Run(displayHost bool) (output string, finalErr error) {
 	case Get, GetID:
 		err = mf.handleGet()
 
-	case Put, PutID:
-		err = mf.handlePut()
+	case Put:
+		// If mongofiles --put ... is called, i.e. with multiple supporting
+		// arguments, then add gridFiles specified in mf.FileNameList
+		for _, filename := range mf.FileNameList {
```

Comment:
Yeah, I had it implemented that way for consistency the first time around, but @mattChiaravalloti recommended that I do it this way instead. I don't think it'd be a big deal one way or another, in any case.

@mattChiaravalloti Any opinions on this?

Reply:
I disagreed that the loop should be inside `handlePut` since it had this code:
```
n, err := mf.put(id, mf.FileName)
if err != nil { ... }
log("copied n bytes")
log("added mf.FileName")
```
duplicated in the body of the method. It appeared in a loop if `mf.FileNameList` was non-empty and outside the loop if it _was_ empty.

Personally, I still think the abstraction as-is makes more sense (`handlePut` handles putting a single file), but if you want to combine the `FileNameList` and `FileName` cases into one that's ok. I advise doing it this way instead of how you initially had it:
```
func (mf) handlePut() (...) {
    id, err := mf.parseOrCreateID()
    if err != nil { ... }

    if len(mf.FileNameList) == 0 {
        mf.FileNameList = []string{mf.FileName}
    }

    for _, filename := range mf.FileNameList {
        n, err := mf.put(id, filename)
        if err != nil { ... }
        log("...")
    }
    ...
}
```
This way, instead of having the same code in two branches of a conditional, you make the simplest conditional possible so the rest of the code follows a single path.

Reply:
Yeah I'm okay with the abstraction being for either a single or multiple files. But whatever it is I think it should be consistent with `handleGet()`. Right now it isn't consistent and that feels a bit messy to me.

Reply:
I wouldn't be against say adding a handleGetID and handlePutID if this helps clean things up. I'm fine with any approach as long as it's consistent.

Reply:
I see. Making it consistent with `handleGet` sounds good. No need to start introducing new methods yet. @hariamoor just try to avoid duplicate code inside the body of `handlePut`!

Reply:
@mattChiaravalloti Followed your suggestion in [be873f0](https://github.com/mongodb/mongo-tools/pull/283/commits/be873f0987be8536e79abc46da18bf9254b9f907).

Reply:
Yes, thank you. It looks good 👍 Good thing we caught the issue of having the `id` logic in the loop (since my suggestion above is incorrect).

Reply:
Note: Matt and I came to the understanding off-band that each file should have a unique ID. That's why I deviate from his suggestion and put `parseOrCreateID` in the loop here.